calico_containers/tests/st/test_add_container.py

```diff
@@ -0,0 +1,25 @@
+from sh import ErrorReturnCode
+from functools import partial
+
+from test_base import TestBase
+from docker_host import DockerHost
+
+
+class TestAddContainer(TestBase):
+    def test_add_container(self):
+        host = DockerHost('host')
+
+        calicoctl = "/code/dist/calicoctl %s"
+
+        host.execute("docker run -tid --name=node busybox")
+        host.execute(calicoctl % "node --ip=127.0.0.1")
+        host.execute(calicoctl % "profile add TEST_GROUP")
+
+        self.assert_powerstrip_up(host)
+
+        host.execute(calicoctl % "container add node 192.168.1.1")
+        host.execute(calicoctl % "profile TEST_GROUP member add node")
+
+        # Wait for felix to program down the route.
+        powerstrip = partial(host.execute, "ip route | grep '192\.168\.1\.1'")
+        assert self.retry_until_success(powerstrip, ex_class=ErrorReturnCode)
```

Comment:
I considered it; but ATM it's only used in this one test, so we might want to wait until it proves itself useful before functioning out. (We might even want to change this to a ping assertion instead.)